test/test_journalpump.py

```diff
@@ -358,25 +359,66 @@ def test_blacklist(self):
         assert data == {"Foo": "a", "_bar": "b", "_zob": "c"}
 
 
+class TestUnitLogLevels(TestCase):
+    def test_empty(self):
+        log_level = "INFO"
+        ll = UnitLogLevel("test", {"unit-a": log_level})
+        data = {"Foo": "bar", "PRIORITY": LOG_SEVERITY_MAPPING[log_level], "SYSTEMD_UNIT": "unit-b"}
+        assert ll.filter_by_level(data) == {}
+
+    def test_matching_level(self):
+        log_level = "INFO"
+        ll = UnitLogLevel("test", {"unit-a": log_level})
+        data = {"Foo": "bar", "PRIORITY": LOG_SEVERITY_MAPPING[log_level], "SYSTEMD_UNIT": "unit-a"}
+        assert ll.filter_by_level(data) == data
+
+    def test_higher_level(self):
+        log_level = "WARNING"
+        ll = UnitLogLevel("test", {"unit-a": log_level})
+        data = {"Foo": "bar", "PRIORITY": LOG_SEVERITY_MAPPING[log_level] + 1, "SYSTEMD_UNIT": "unit-a"}
+        assert ll.filter_by_level(data) == {}
+
+    def test_lower_level(self):
+        log_level = "WARNING"
+        ll = UnitLogLevel("test", {"unit-a": log_level})
+        data = {"Foo": "bar", "PRIORITY": LOG_SEVERITY_MAPPING[log_level] - 1, "SYSTEMD_UNIT": "unit-a"}
+        assert ll.filter_by_level(data) == data
```

Comment:
Could we have a couple of test cases to match when `SYSTEMD_UNIT` and/or `PRIORITY` are not present in the entry? We should be allowing these though regardless of the filter.

Reply:
We can indeed! Added